### PR TITLE
random forest and tensorflow

### PR DESCRIPTION
**Pull request by muskanstha**

### CODE REVIEW

File: index.py [hunk 1]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
 print('Variance score: %.2f' % r2_score(yTest, yPred))
 
 
+#%%
+
+# =============================================================================
+# 7. Serialize processed dataset
+# =============================================================================
+
+# import pickle
```
Comment (pixochi) [inline index.py:308]: Is this commented section needed?

File: index.py [hunk 2]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+                               criterion='mse', 
+                               random_state=3, 
+                               n_jobs=-1)
+forest.fit(XTrain, yTrain)
+
+
+
+#%%
+y_train_pred = forest.predict(XTrain)
+y_test_pred = forest.predict(XTest)
```
Comment (pixochi) [inline index.py:396]: I think predictions should be made only on XTest  because XTtrain is used to fit the model.

File: index.py [hunk 3]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+# =============================================================================
+
+import tensorflow as tf
+from tensorflow import keras
+from tensorflow.keras import layers
+
+#%%
+def build_model():
+  model = keras.Sequential([
+    # layers.Dense(64, activation=tf.nn.relu, input_shape=[77]),
```
Comment (pixochi) [inline index.py:420]: Dead code?

**Review by pixochi — CHANGES_REQUESTED** [review]

**Comment by pixochi** [comment]
@muskanstha would you mind removing the comments for notebook cells?

Because Spyder highlights the cells with this ugly and distracting color.
image

+ plus you have to run the cells 1 by 1

**Comment by marcinzelent** [comment]
@pixochi First world problems.

**Comment by pixochi** [comment]
> @pixochi First world problems.

@marcinzelent  A first world problem is still a problem.
- Mahatma Gandhi

File: index.py [hunk 4]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+
+# =============================================================================
+# 8. VISUALIZATIONS
+# =============================================================================
+
+import matplotlib.pyplot as plt
+
+
+#%%
+y_label.hist(bins=50, figsize=(5,5))
```
Comment (pixochi) [inline index.py:343]: Why 50 bins? it gives you 1 bin for the price range of 40usd. Change it to 40 or 20 to get a range of 50,100usd respectively. You could also use the `width` parameter with value 40 or 90, again respectively, to make it more readable.

File: index.py [hunk 5]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+plt.tight_layout()
+
+
+#%%
+y_label.plot(kind='box', showfliers=False, subplots=True, layout=(1,12),figsize=(20,7), sharex=False, sharey=False)
+plt.tight_layout()
+
+
+#%%
+plt.scatter(dataset_processed['beds'],dataset_processed['bedrooms'])
```
Comment (pixochi) [inline index.py:358]: Is this helpful?...it looks like there are listings with 0 beds and 5 bedrooms, could it be correct?
image

File: index.py [hunk 6]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+#%%
+import seaborn as sns
+
+plt.figure(figsize=(15,8))
+corr = dataset_processed[["price","bedrooms","beds",'minimum_nights','maximum_nights']].corr()
+mask = np.zeros_like(corr, dtype=np.bool)
+mask[np.triu_indices_from(mask)] = True
+sns.heatmap(corr, linewidths=.5,annot=True,mask=mask,cmap='coolwarm')
+#%%
+sns.pairplot(dataset_processed[["price","bedrooms","beds",'minimum_nights','maximum_nights']], diag_kind="kde")
```
Comment (pixochi) [inline index.py:379]: Could you add a title for the graph?

File: index.py [hunk 7]
```diff
@@ -288,6 +299,237 @@ def backwardEliminateFeatures(features, y_label, p_threshold):
...
+# =============================================================================
+# 9. RANDOM FOREST MODEL
+# =============================================================================
+# create and train the model
+from sklearn.ensemble import RandomForestRegressor
+forest = RandomForestRegressor(n_estimators=500, 
+                               criterion='mse', 
+                               random_state=3, 
+                               n_jobs=-1)
+forest.fit(XTrain, yTrain)
```
Comment (pixochi) [inline index.py:390]: Getting this on forest.fit

image

**Review by pixochi — CHANGES_REQUESTED** [review]

File: index.py (hunk 5 above)
Comment (muskanstha) [reply index.py]: image

Comment (pixochi) [reply index.py:358]: @muskanstha  In the picture `beds` are 0 and `bedrooms` 5

File: index.py (hunk 3 above)
Comment (muskanstha) [reply index.py]: ?

File: index.py (hunk 2 above)
Comment (muskanstha) [reply index.py:396]: it is to check the result of the model on known vs unknown data

File: index.py (hunk 7 above)
Comment (muskanstha) [reply index.py]: No idea, working here!
image

File: index.py (hunk 3 above)
Comment (pixochi) [reply index.py:420]: the commented code

File: index.py (hunk 2 above)
Comment (pixochi) [reply index.py:396]: is it a common technique to test the model on known data?

File: index.py (hunk 7 above)
Comment (pixochi) [reply index.py:390]: It's the 1st line in your console

**Pull request closed without merging**